addons/udes_stock/tests/test_refactoring_splitting.py

```diff
@@ -923,3 +923,75 @@ def test_split_mixed_products_partial_reserve(self):
         ]
         expected_move_format = [(self.apple, 2, 0), (self.banana, 3, 0)]
         self.assertCountEqual(move_format, expected_move_format)
+
+class TestRefactoringDateDone(common.BaseUDES):
+    @classmethod
+    def setUpClass(cls):
+        """
+        Create stock: pallet with apples, pallet with bananas
+        create picking: for all of both
+        """
+        super().setUpClass()
+
+        # putaway is being refactored
+        cls.picking_type_putaway.write(
+            {
+                "u_post_assign_action": "group_by_move_line_key",
+                "u_move_line_key_format": "{package_id.name}",
+            }
+        )
+
+        cls.picking = cls.create_picking(cls.picking_type_in)
+
+        # Create two Goods In pickings
+        product_info_1 = [{"product": cls.apple, "qty": 5}]
+        product_info_2 = [{"product": cls.apple, "qty": 10}, {"product": cls.banana, "qty": 10}]
+
+        cls.pick_1 = cls.create_picking(cls.picking_type_in, products_info=product_info_1)
+        cls.pick_2 = cls.create_picking(cls.picking_type_in, products_info=product_info_2)
+
+    def test_does_not_propagate_date_done_to_incomplete_picking(self):
+        """Check the done date is not carried over to incomplete pickings"""
+        # Complete pick_1 and assert put away is generated and then assert that date done is set on the Goods In but not on the put away
+        self.pick_1.action_assign()
```

Comment:
You can probably assign in the create function

Reply:
Just to clarify, do you mean the setup function?

Reply:
you can pass assign=True as a kwarg in the create_picking function so we don't need to call this seperately.